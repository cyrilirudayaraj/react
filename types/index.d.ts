import { FormikProps } from 'formik';

interface BaseType {
  id: string;
  name: string;
  description: string;
}
export interface NewTask {
  name: string;
  priorityId: string;
  priorityReasonId: string;
  priorityReasonNote: string;
  originatingSystemId: string;
  originatingCaseId: string;
  taskTypeId: string;
  legacyRuleId: string;
  legacyTaskId: string;
  businessRequirementName: string;
  businessRequirementId: string;
  dueDate: string;
  assignedTo: string;
  businessRequirementValue: string;
  departmentOriginId: string;
  brUpdateReasonIds: any[];
  clientDueDate?: string;
  isArchivedRule?: boolean;
  isLegacyRuleExist?: boolean;
}

export interface TaskStep {
  workflowStepId: any;
  signOffStatement: any;
  signedOffYn: any;
  name: string;
  ordering: string;
  transitionText: any;
  id: string;
  assignedTo: string | null;
  version: string;
  teamId?: string;
  plannedCompletion: string | null;
  actualCompletion: string | null;
  totalWorkLog?: string;
}

export interface TaskDTDetails {
  id: string;
  taskId: string;
  modelName: string;
  name: string;
  refId: string;
  description: string;
  refUrl: string;
  ordering?: string;
  lastModified?: string;
  lastModifiedBy?: string;
  deleted?: string;
  deletedBy?: string;
  version: string;
  expanded?: boolean;
}

export interface TaskDetail {
  originatingSystemId: string | '';
  statusId: string;
  dueDate: any;
  status: string;
  createdBy: string;
  contextId: string;
  contextName: string;
  contextRefId: string;
  ruleReportingCategoryId: any;
  businessRequirementId: string;
  originatingCaseId: string;
  clientDueDate: any;
  modelDesignChanges: any;
  localRuleUseCaseId: any;
  id: string;
  version: string;
  activeTaskStepId: string;
  activeWorkflowStepId: string;
  businessRequirementTypeId: any;
  priorityName: string;
  dtChangesYn: any;
  legacyRuleId: string;
  jiraId: string | null;
  legacyTaskId: string | null;
  legacyRuleArchived: string | null;
  businessRequirementName: string;
  name: string;
  taskTypeId: string;
  taskTypeName: string;
  priorityId: string;
  description: any;
  taskSteps: TaskStep[];
  created: string;
  lastModified: string;
  businessRequirementDesc: any;
  athenaNetChangesYn: any;
  lastModifiedBy: string;
  deploymentDate: string;
  version: string;
  testClaimExample: any;
  testNote: any;
  modelBranchUrl: string;
  dtBranchUrl: string;
  taskDecisionTableDetails: TaskDTDetails[];
  taskDependencies: TaskDependencyDetails[];
  originatingSystemName: string;
  escalatedYn: string;
  associatedBRTasks: TaskInfo[];
  canSignOff?: any;
  ruleEngineId: any;
  ruleTypeId: any;
  visitRuleDisplayLocationId: any;
  internalFixText1: string;
  internalFixText2: string;
  rules2TransformationStatusId: number;
  departmentOriginId: number;
  departmentOriginName: string;
  brUpdateReasons: any[];
  brUpdateReasonIds: any[];
  claimRuleCategoryId: string | null;
  scrubTypeId: string;
  claimRuleCategoryRefId: string;
  claimRuleCategory: string;
  rtSynchronizationLog: any;
  priorityReasonId: any;
  priorityReason: string;
  priorityReasonNote: any;
  rejectionReason: string;
  rejectionDescription: string;
}

export interface DependencyDetails {
  dependencyId: string;
  dependencyName: string;
  dependencySystemId: string;
  deploymentDate: string;
  statusName: string;
  activeWorkflowStepId?: string;
}

export interface TaskDependencyDetails {
  dependencyDate: string;
  dependencyId: string;
  dependencyName: string;
  dependencySystemId: string;
  dependencySystemName: string;
  deploymentDate: string;
  id: string;
  notes: string;
  ordering: string;
  statusName: string;
  taskId: string;
  dependencyCondition: string;
  version: string;
  dependencySystemId: string;
  completedYn: string;
  description: string;
  deleted?: string;
}
export interface TaskInfo {
  id: string;
  name: string;
  activeTaskStepId: string;
  activeTaskStepName: string;
}
export interface TaskType {
  id: string;
  name: string;
  description: string;
}

export interface ReportType {
  id: string;
  name: string;
  description: string;
}

export interface Task {
  activeTaskStepId: string;
  assignedTo: string | null;
  businessRequirementId: string;
  dueOn: string;
  lastModified: string;
  legacyRuleId: string;
  priorityName: string;
  statusName: string;
  taskId: string;
  taskName: string;
  taskTypeId: string;
  taskTypeName: string;
  workflowStepName: string;
  escalatedYn: string;
}

export interface UpdateTaskProps {
  task: TaskDetail;
  taskUpdateAction?: fn;
  toggleEdit?: fn;
  editBtnClick?: any;
  editBtnValue?: any;
  getEditBtnValue?: any;
  leftSectionActive?: any;
  rightSectionActive?: any;
  setLeftFormAction?: fn;
  setRightFormAction?: fn;
}

export type ToastAlertTypes = 'attention' | 'info' | 'success';

export interface ToastMessage {
  id?: string;
  headerText: string | fn;
  alertType?: ToastAlertTypes;
  message: string | fn;
  params?: Record<string, unknown>;
}

export interface ReportCriteria {
  reportTypeId: string;
  dateType: string;
  startDate: date;
  endDate: date;
}

export interface UserComment {
  id: string;
  taskId: string;
  content: any;
  created: string;
  createdBy: string;
  createdTimestamp: string;
}

export interface TaskDiscusisonProps {
  task: TaskDetail;
  fetchUserCommentDetails?: fn;
  taskId: any;
  commentId: any;
  resetCommentId: any;
}

export interface DiscussionFormProps {
  taskdetails: TaskDetail;
  createUserComment: fn;
  comments: UserComment[];
  resetPopupActions?: any;
  resetCommentId: any;
}

export interface UserCommentFormProps {
  formik: FormikProps<any>;
}

export type BusinessRequirementType = BaseType;

export interface RuleReportingCategory extends BaseType {
  refId: string;
}

export interface Context extends BaseType {
  refId: string;
}

export type LocalRuleUseCase = BaseType;

export interface AttachmentDetail {
  id: string;
  taskId: string;
  taskStepId: any;
  description: string;
  attachmentTypeId: string;
  attachmentName: string;
  createdBy: string;
  fileName: string;
  filePath: string;
  version: string;
}

export interface TaskDocumentProps {
  task: TaskDetail;
  fetchAttachmentDetails?: fn;
}

export interface AttachmentType {
  id: string;
  name: string;
  version: string;
}
export type WorkFlowFilterType = BaseType;

export interface StatusFilterType {
  workFlowStepQueueIds: string;
  name: string;
  id: string;
}
export interface TimeEntry {
  id: string;
  taskId: string;
  taskStepId: any;
  spentOn: string;
  spentTimeInMins: string;
  workflowStepName: string;
  workflowStepId: string;
  createdBy: string;
  created: string;
  version: string;
  totalSpentTimeInMins: string;
}
export interface WorkLogProps {
  task: TaskDetail;
  fetchTimeEntryLogs?: fn;
}

export interface AuditEntry {
  fieldName?: string;
  oldValue?: Record<string, string | null> | string | null;
  newValue?: Record<string, string | null> | string | null;
}

export interface AuditEvent {
  eventType: string;
  objectType: string;
  eventName: string;
  created: string;
  createdBy: string;
  changes: AuditEntry[];
}

export interface Shape {
  text: string;
  value: string;
}

export interface Deployment {
  branchName: string;
  created: string;
  createdBy: string;
  dcUrl: string;
  decisionServiceName: string;
  deploymentConfigName: string;
  description: string | null;
  id: any;
  name: string;
  releaseVersion: string | null;
  status: string;
  statusId: string;
  verifiedYn: string | null;
  version: any;
  archiveFileName: string;
  archiveFilePath: string;
  envNames?: string;
  taskIds?: string;
}

export interface DeploymentLog {
  action: string;
  category: string | null;
  id: string;
  message: string;
  name: string;
  ordering: string;
  request: string;
  response: string;
  statusId: string;
  version: string;
}

export interface DeploymentEnv {
  id: string;
  initialVersion: string;
  name: string;
  resUrl: string;
  resultVersion: string;
  statusId: string;
  version: string;
}

export interface DeploymentTask {
  businessRequirementId: string;
  businessRequirementName: string;
  id: string;
  legacyRuleId: string;
  name: string;
  taskTypeId: string;
  taskTypeName: string;
}

export interface DeploymentDetails extends Deployment {
  logs: DeploymentLog[];
  tasks: DeploymentTask[];
  envs: DeploymentEnv[];
}

export interface DepartmentOrigin {
  id: string;
  name: string;
  description: string;
}

export interface BRUpdateReason {
  id: string;
  name: string;
  description: string;
}

export interface DeploymentStatus {
  ordering: string;
  name: string;
  id: string;
}

export interface PriorityPolicy {
  description1: string;
  description2: string;
  externalDueDate: string;
  externalTatInDays: string;
  id: string;
  internalDueDate: string;
  internalTatInDays: string;
  name: string;
}

export interface RejectionReason {
  id: string;
  name: string;
  version: string;
  ordering: string;
}
export interface RulesMigrationDetail {
  id: string;
  ruleId: string;
  contextId: string;
  businessRequirementId: string;
  migrationStatus: string;
  comment: string;
}

export interface BRDetails {
  legacyRuleId: string;
  rules2TransformationStatusId: string;
  rules2TransformationStatus: string;
  name: string;
  ruleReportingCategoryId: string;
  ruleReportingCategoryName: string;
  description: string;
  internalFixText2: string;
  ruleTypeId: string;
  ruleType: string;
  internalFixText1: string;
  localRuleUseCaseId: string;
  localRuleUseCaseName: string;
  contextId: string;
  contextName: string;
  id: string;
  visitRuleDisplayLocationId: string;
  visitRuleDisplayLocation: string;
  businessRequirementTypeId: string;
  businessRequirementType: string;
  associatedBRTasks: any[];
}
