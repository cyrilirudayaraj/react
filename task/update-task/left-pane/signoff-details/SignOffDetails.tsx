import React from 'react';
import { Table } from '@athena/forge';
import { TaskDetail } from '../../../../../types';
import Labels from '../../../../../constants/Labels';
import StringUtil from '../../../../../utils/StringUtil';

const labels = Labels.TASK_SIGNOFF_DETAILS;
interface SignOffDetailsProps {
  task?: TaskDetail;
}

export default function SignOffDetails(
  props: SignOffDetailsProps
): JSX.Element {
  const dataSource = props.task?.taskSteps ? props.task.taskSteps : [];

  const getColumnTemplate = (value: string) => {
    return value ? value : labels.HYPHEN;
  };

  const getSignedOffByTemplate = (value: string, { rowData }: any) => {
    return value &&
      !StringUtil.isNullOrEmpty(rowData.signedOffYn) &&
      rowData.signedOffYn.toLowerCase() === labels.TXT_YES
      ? value
      : labels.HYPHEN;
  };
  return (
    <div className="sign-off">
      <div className="panel">
        <div className="panel-content signoff-content">
          <div className="fe_u_flex-container">
            <Table
              layout="compact"
              className="full-width"
              rows={dataSource}
              rowKey="id"
              columns={[
                {
                  key: 'name',
                  displayName: labels.WORKFLOW_STEP,
                },
                {
                  key: 'plannedCompletion',
                  displayName: labels.NEXT_ACTION_DUE,
                  template: getColumnTemplate,
                },
                {
                  key: 'assignedTo',
                  displayName: labels.SIGNED_OFF_BY,
                  template: getSignedOffByTemplate,
                },
                {
                  key: 'actualCompletion',
                  displayName: labels.SIGNED_OFF_DATE,
                  template: getColumnTemplate,
                },
              ]}
            />
          </div>
        </div>
      </div>
    </div>
  );
}
