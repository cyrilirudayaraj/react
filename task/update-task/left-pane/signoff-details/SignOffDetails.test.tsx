import React from 'react';
import { configure, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { Table } from '@athena/forge';
import SignOffDetails from './SignOffDetails';
import { TaskDetail } from '../../../../../types';

configure({ adapter: new Adapter() });

describe('test <SignOffDetails>', () => {
  let wrapper: any;
  const taskDetails = {} as TaskDetail;
  taskDetails.taskSteps = [
    {
      workflowStepId: null,
      signOffStatement: null,
      signedOffYn: 'y',
      name: 'Analyze Task',
      ordering: '1',
      transitionText: null,
      id: '101',
      assignedTo: 'vsivachandran',
      version: '1',
      plannedCompletion: '01/01/2000',
      actualCompletion: '01/01/2000',
    },
    {
      workflowStepId: null,
      signOffStatement: null,
      signedOffYn: null,
      name: 'Review Task',
      ordering: '2',
      transitionText: null,
      id: '102',
      assignedTo: null,
      version: '1',
      plannedCompletion: null,
      actualCompletion: null,
    },
    {
      workflowStepId: null,
      signOffStatement: null,
      signedOffYn: null,
      name: 'Manager Review',
      ordering: '3',
      transitionText: null,
      id: '103',
      assignedTo: null,
      version: '1',
      plannedCompletion: null,
      actualCompletion: null,
    },
    {
      workflowStepId: null,
      signOffStatement: null,
      signedOffYn: null,
      name: 'Make Model Changes',
      ordering: '4',
      transitionText: null,
      id: '104',
      assignedTo: null,
      version: '1',
      plannedCompletion: null,
      actualCompletion: null,
    },
    {
      workflowStepId: null,
      signOffStatement: null,
      signedOffYn: 'y',
      name: 'Make Code Changes',
      ordering: '5',
      transitionText: null,
      id: '105',
      assignedTo: null,
      version: '1',
      plannedCompletion: null,
      actualCompletion: null,
    },
    {
      workflowStepId: null,
      signOffStatement: null,
      signedOffYn: null,
      name: 'Test Changes',
      ordering: '6',
      transitionText: null,
      id: '106',
      assignedTo: null,
      version: '1',
      plannedCompletion: null,
      actualCompletion: null,
    },
    {
      workflowStepId: null,
      signOffStatement: null,
      signedOffYn: null,
      name: 'Review Changes',
      ordering: '7',
      transitionText: null,
      id: '107',
      assignedTo: null,
      version: '1',
      plannedCompletion: null,
      actualCompletion: null,
    },
    {
      workflowStepId: null,
      signOffStatement: null,
      signedOffYn: null,
      name: 'Deploy Changes',
      ordering: '8',
      transitionText: null,
      id: '108',
      assignedTo: null,
      version: '1',
      plannedCompletion: null,
      actualCompletion: null,
    },
  ];
  beforeEach(() => {
    wrapper = mount(<SignOffDetails task={taskDetails} />);
  });

  it('should render as expected', () => {
    expect(wrapper.debug()).toMatchSnapshot();
  });

  it('should render Table element', () => {
    expect(wrapper.find(Table)).toHaveLength(1);
  });

  it('should render as expected with empty data source', () => {
    let task;
    wrapper = mount(<SignOffDetails task={task} />);
    expect(wrapper.find(Table)).toHaveLength(1);
  });
});
