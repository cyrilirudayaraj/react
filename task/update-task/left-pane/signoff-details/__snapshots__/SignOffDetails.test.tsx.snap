// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test <SignOffDetails> should render as expected 1`] = `
"<SignOffDetails task={{...}}>
  <div className=\\"sign-off\\">
    <div className=\\"panel\\">
      <div className=\\"panel-content signoff-content\\">
        <div className=\\"fe_u_flex-container\\">
          <Table layout=\\"compact\\" className=\\"full-width\\" rows={{...}} rowKey=\\"id\\" columns={{...}} showHover={true} onRow={[Function: onRow]} replaceChildren={false} nowrapHeaders={false}>
            <Provider className=\\"fe_c_table fe_c_table--show-hover fe_c_table--compact full-width\\" columns={{...}} renderers={{...}}>
              <table className=\\"fe_c_table fe_c_table--show-hover fe_c_table--compact full-width\\">
                <Header>
                  <thead>
                    <HeaderRow renderers={{...}} onRow={[Function: onRow]} rowData={{...}} rowIndex={0}>
                      <tr>
                        <th data-column=\\"name\\" scope=\\"col\\" role=\\"columnheader\\" title={{...}} className=\\"fe_c_table__header\\">
                          Workflow Step
                        </th>
                        <th data-column=\\"plannedCompletion\\" scope=\\"col\\" role=\\"columnheader\\" title={{...}} className=\\"fe_c_table__header\\">
                          Next Action Due
                        </th>
                        <th data-column=\\"assignedTo\\" scope=\\"col\\" role=\\"columnheader\\" title={{...}} className=\\"fe_c_table__header\\">
                          Signed-off-by
                        </th>
                        <th data-column=\\"actualCompletion\\" scope=\\"col\\" role=\\"columnheader\\" title={{...}} className=\\"fe_c_table__header\\">
                          Signed-off Date
                        </th>
                      </tr>
                    </HeaderRow>
                  </thead>
                </Header>
                <Body onRow={[Function: onRow]} rows={{...}} rowKey=\\"id\\">
                  <tbody>
                    <BodyRow renderers={{...}} onRow={[Function: onRow]} rowKey=\\"101-row\\" rowIndex={0} rowData={{...}} columns={{...}}>
                      <tr>
                        <td data-column=\\"name\\" className=\\"fe_c_table__cell\\">
                          Analyze Task
                        </td>
                        <td data-column=\\"plannedCompletion\\" className=\\"fe_c_table__cell\\">
                          01/01/2000
                        </td>
                        <td data-column=\\"assignedTo\\" className=\\"fe_c_table__cell\\">
                          vsivachandran
                        </td>
                        <td data-column=\\"actualCompletion\\" className=\\"fe_c_table__cell\\">
                          01/01/2000
                        </td>
                      </tr>
                    </BodyRow>
                    <BodyRow renderers={{...}} onRow={[Function: onRow]} rowKey=\\"102-row\\" rowIndex={1} rowData={{...}} columns={{...}}>
                      <tr>
                        <td data-column=\\"name\\" className=\\"fe_c_table__cell\\">
                          Review Task
                        </td>
                        <td data-column=\\"plannedCompletion\\" className=\\"fe_c_table__cell\\">
                          -
                        </td>
                        <td data-column=\\"assignedTo\\" className=\\"fe_c_table__cell\\">
                          -
                        </td>
                        <td data-column=\\"actualCompletion\\" className=\\"fe_c_table__cell\\">
                          -
                        </td>
                      </tr>
                    </BodyRow>
                    <BodyRow renderers={{...}} onRow={[Function: onRow]} rowKey=\\"103-row\\" rowIndex={2} rowData={{...}} columns={{...}}>
                      <tr>
                        <td data-column=\\"name\\" className=\\"fe_c_table__cell\\">
                          Manager Review
                        </td>
                        <td data-column=\\"plannedCompletion\\" className=\\"fe_c_table__cell\\">
                          -
                        </td>
                        <td data-column=\\"assignedTo\\" className=\\"fe_c_table__cell\\">
                          -
                        </td>
                        <td data-column=\\"actualCompletion\\" className=\\"fe_c_table__cell\\">
                          -
                        </td>
                      </tr>
                    </BodyRow>
                    <BodyRow renderers={{...}} onRow={[Function: onRow]} rowKey=\\"104-row\\" rowIndex={3} rowData={{...}} columns={{...}}>
                      <tr>
                        <td data-column=\\"name\\" className=\\"fe_c_table__cell\\">
                          Make Model Changes
                        </td>
                        <td data-column=\\"plannedCompletion\\" className=\\"fe_c_table__cell\\">
                          -
                        </td>
                        <td data-column=\\"assignedTo\\" className=\\"fe_c_table__cell\\">
                          -
                        </td>
                        <td data-column=\\"actualCompletion\\" className=\\"fe_c_table__cell\\">
                          -
                        </td>
                      </tr>
                    </BodyRow>
                    <BodyRow renderers={{...}} onRow={[Function: onRow]} rowKey=\\"105-row\\" rowIndex={4} rowData={{...}} columns={{...}}>
                      <tr>
                        <td data-column=\\"name\\" className=\\"fe_c_table__cell\\">
                          Make Code Changes
                        </td>
                        <td data-column=\\"plannedCompletion\\" className=\\"fe_c_table__cell\\">
                          -
                        </td>
                        <td data-column=\\"assignedTo\\" className=\\"fe_c_table__cell\\">
                          -
                        </td>
                        <td data-column=\\"actualCompletion\\" className=\\"fe_c_table__cell\\">
                          -
                        </td>
                      </tr>
                    </BodyRow>
                    <BodyRow renderers={{...}} onRow={[Function: onRow]} rowKey=\\"106-row\\" rowIndex={5} rowData={{...}} columns={{...}}>
                      <tr>
                        <td data-column=\\"name\\" className=\\"fe_c_table__cell\\">
                          Test Changes
                        </td>
                        <td data-column=\\"plannedCompletion\\" className=\\"fe_c_table__cell\\">
                          -
                        </td>
                        <td data-column=\\"assignedTo\\" className=\\"fe_c_table__cell\\">
                          -
                        </td>
                        <td data-column=\\"actualCompletion\\" className=\\"fe_c_table__cell\\">
                          -
                        </td>
                      </tr>
                    </BodyRow>
                    <BodyRow renderers={{...}} onRow={[Function: onRow]} rowKey=\\"107-row\\" rowIndex={6} rowData={{...}} columns={{...}}>
                      <tr>
                        <td data-column=\\"name\\" className=\\"fe_c_table__cell\\">
                          Review Changes
                        </td>
                        <td data-column=\\"plannedCompletion\\" className=\\"fe_c_table__cell\\">
                          -
                        </td>
                        <td data-column=\\"assignedTo\\" className=\\"fe_c_table__cell\\">
                          -
                        </td>
                        <td data-column=\\"actualCompletion\\" className=\\"fe_c_table__cell\\">
                          -
                        </td>
                      </tr>
                    </BodyRow>
                    <BodyRow renderers={{...}} onRow={[Function: onRow]} rowKey=\\"108-row\\" rowIndex={7} rowData={{...}} columns={{...}}>
                      <tr>
                        <td data-column=\\"name\\" className=\\"fe_c_table__cell\\">
                          Deploy Changes
                        </td>
                        <td data-column=\\"plannedCompletion\\" className=\\"fe_c_table__cell\\">
                          -
                        </td>
                        <td data-column=\\"assignedTo\\" className=\\"fe_c_table__cell\\">
                          -
                        </td>
                        <td data-column=\\"actualCompletion\\" className=\\"fe_c_table__cell\\">
                          -
                        </td>
                      </tr>
                    </BodyRow>
                  </tbody>
                </Body>
              </table>
            </Provider>
          </Table>
        </div>
      </div>
    </div>
  </div>
</SignOffDetails>"
`;
